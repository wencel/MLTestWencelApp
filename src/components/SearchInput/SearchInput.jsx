import PropTypes from 'prop-types';
import Styles from './SearchInput.module.sass';
import glassImg from 'assets/img/mlGlass.png';

const SearchInput = ({
  onSearch,
  inputProps,
  buttonProps,
  value,
  onChange,
}) => {
  const submitForm = e => {
    e.preventDefault();
    onSearch(value);
  };
  return (
    <form className={Styles.SearchInput} onSubmit={submitForm}>
      <input {...inputProps} value={value} onChange={onChange} />
      <button {...buttonProps}>
        <img src={glassImg} alt='Search Img' />
      </button>
    </form>
  );
};

SearchInput.defaultProps = {
  onSearch: () => {},
  buttonProps: {},
  inputProps: {},
};

SearchInput.propTypes = {
  buttonProps: PropTypes.object,
  inputProps: PropTypes.object,
  onChange: PropTypes.func,
  onSearch: PropTypes.func,
  value: PropTypes.string,
};

export default SearchInput;
