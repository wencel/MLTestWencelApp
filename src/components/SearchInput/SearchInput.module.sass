@import  'styles.global.sass'
.SearchInput
  width: 100%
  height: 100%
  display: flex
  input
    width: 100%
    height: 100%
    border-top-left-radius: 5px
    border-bottom-left-radius: 5px
    border-top-right-radius: 0
    border-bottom-right-radius: 0
    border: none
    padding: 0 1em
    font-size: 1em
  button
    background: $bgColor
    width: 3em
    height: 100%
    border-top-left-radius: 0
    border-bottom-left-radius: 0
    border-top-right-radius: 5px
    border-bottom-right-radius: 5px
    border: none
    padding: 0.7em
    margin: 0
    cursor: pointer
    &:active
      background-color: $darkBgColor
    img
      width: 100%

    
