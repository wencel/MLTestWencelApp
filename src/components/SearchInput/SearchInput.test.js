import React from 'react';
import { mount } from 'enzyme';
import SearchInput from './SearchInput';

describe('WHEN mounting the component', () => {
  const onSearch = jest.fn(0);
  const onChange = jest.fn(0);
  const wrapper = mount(
    <SearchInput onChange={onChange} onSearch={onSearch} value='Value' />
  );
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('calls the onSearch function when subbmitting the form', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(onSearch).toHaveBeenCalledTimes(1);
  });
});
