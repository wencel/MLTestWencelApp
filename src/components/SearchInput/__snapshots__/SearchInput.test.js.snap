// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `ReactWrapper {}`;
